libraries/AP_Motors/AP_Motors_Class.h

```diff
@@ -64,6 +68,9 @@ class AP_Motors {
         MOTOR_FRAME_TYPE_BF_X_REV = 18, // X frame, betaflight ordering, reversed motors
     };
 
+    // return string corresponding to frame_type
+    virtual const char* get_type_string() { return ""; }
```

Comment:
Perhaps return something distinctive here?

Also, if this is a can't-happen, please put in an internal_error::flow_of_control

Reply:
some frame classes don't have types, so an empty string is appropriate, but perhaps "unused" would be better?
If frame_type is significant, then this should be overridden with something more appropriate.
I'm not sure an internal error would be a good idea, since this is a public method and we have no control over who's calling it.